src/proxy/ReplicatedMapProxy.ts

```diff
@@ -0,0 +1,201 @@
+import {ReplicatedMapRemoveCodec} from '../codec/ReplicatedMapRemoveCodec';
+import {ReplicatedMapPutAllCodec} from '../codec/ReplicatedMapPutAllCodec';
+import {ReplicatedMapKeySetCodec} from '../codec/ReplicatedMapKeySetCodec';
+import {ReplicatedMapValuesCodec} from '../codec/ReplicatedMapValuesCodec';
+import {ReplicatedMapEntrySetCodec} from '../codec/ReplicatedMapEntrySetCodec';
+import {Predicate} from '../core/Predicate';
+import {IMapListener} from '../core/MapListener';
+import {ReplicatedMapRemoveEntryListenerCodec} from '../codec/ReplicatedMapRemoveEntryListenerCodec';
+import {EntryEventType} from '../core/EntryEventType';
+import ClientMessage = require('../ClientMessage');
+/* tslint:disable:max-line-length */
+import {ReplicatedMapAddEntryListenerToKeyWithPredicateCodec} from '../codec/ReplicatedMapAddEntryListenerToKeyWithPredicateCodec';
+import {ReplicatedMapAddEntryListenerToKeyCodec} from '../codec/ReplicatedMapAddEntryListenerToKeyCodec';
+import {ReplicatedMapAddEntryListenerWithPredicateCodec} from '../codec/ReplicatedMapAddEntryListenerWithPredicateCodec';
+import {ReplicatedMapAddEntryListenerCodec} from '../codec/ReplicatedMapAddEntryListenerCodec';
+/* tslint:enable:max-line-length */
+
+export class ReplicatedMapProxy<K, V> extends BaseProxy implements IReplicatedMap<K, V> {
+
+    put(key: K, value: V, ttl: number): Promise<V> {
+        assertNotNull(key);
+        assertNotNull(value);
+
+        let valueData: Data = this.toData(value);
+        let keyData: Data = this.toData(key);
+
+        return this.encodeInvokeOnKey<V>(ReplicatedMapPutCodec, keyData, keyData, valueData, ttl);
+    }
+
+    clear(): Promise<void> {
+        return this.encodeInvokeOnRandomTarget<void>(ReplicatedMapClearCodec);
+    }
+
+    get(key: K): Promise<V> {
+        assertNotNull(key);
+
+        const keyData = this.toData(key);
+        return this.encodeInvokeOnKey<V>(ReplicatedMapGetCodec, keyData, keyData);
+    }
+
+    containsKey(key: K): Promise<boolean> {
+        assertNotNull(key);
+
+        const keyData = this.toData(key);
+        return this.encodeInvokeOnKey<boolean>(ReplicatedMapContainsKeyCodec, keyData, keyData);
+    }
+
+    containsValue(value: V): Promise<boolean> {
+        assertNotNull(value);
+
+        const valueData = this.toData(value);
+        return this.encodeInvokeOnRandomPartition<boolean>(ReplicatedMapContainsValueCodec, valueData);
+    }
+
+    size(): Promise<number> {
+        return this.encodeInvokeOnRandomPartition<number>(ReplicatedMapSizeCodec);
+    }
+
+
+    isEmpty(): Promise<boolean> {
+        return this.encodeInvokeOnRandomPartition<boolean>(ReplicatedMapIsEmptyCodec);
+    }
+
+    remove(key: K): Promise<V> {
+        assertNotNull(key);
+
+        const keyData = this.toData(key);
+        return this.encodeInvokeOnKey<V>(ReplicatedMapRemoveCodec, keyData, keyData);
+    }
+
+    putAll(pairs: [K, V][]): Promise<void> {
+        let pair: [K, V];
```

Comment:
Regarding use of `const` and `let` keywords, our typescript compilation setup converts `let` and `const` keywords to `var`s. Use of these keywords may result in hard to spot bugs while maintaining the codebase. Can you use `var` instead.